drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -115,14 +117,14 @@ void ng_at86rf2xx_reset(ng_at86rf2xx_t *dev)
     cpuid[0] &= ~(0x01);
     cpuid[0] |= 0x02;
     /* copy and set long address */
-    memcpy(&addr_long, cpuid, 8);
-    ng_at86rf2xx_set_addr_long(dev, addr_long);
-    /* now compress the long addr to form the short address */
-    for (int i = 2; i < 8; i++) {
-        cpuid[i & 0x01] ^= cpuid[i];
+#if (CPUID_ID_LEN > 8)
+    for (size_t i = 0; i < 8; i++) {
+        cpuid[i] ^= cpuid[i + (CPUID_ID_LEN - 8)];
     }
```

Comment:
ok, just tested with the two iot-labs I have available here: If you remove lines 120 to 124, you should get unique EUIs:
```
node1: 36:32:48:33:46:d5:9f:26
node2: 36:32:48:33:46:d8:9e:16
```